dpytools/http/dataset/base_dataset.py

```diff
@@ -0,0 +1,39 @@
+import json
+from typing import Dict
+
+from requests import RequestException, Response
+
+from dpytools.http.base import BaseHttpClient
+from dpytools.http.upload.token_auth import TokenAuth
+from dpytools.logging.logger import DpLogger
+
+logger = DpLogger("dpytools")
+
+
+class BaseDatasetClient(BaseHttpClient):
+    def __init__(self, url: str, backoff_max: int = 30):
+        super().__init__(backoff_max=backoff_max)
+        self.token_auth = TokenAuth(backoff_max=backoff_max)
+        self.url = url
+        self.dataset_url = f"{self.url}/dataset"
+
+    def send_json(self, url: str, json_data: Dict) -> None:
+        """
+        Send a POST request with JSON data to the specified URL.
+
+        :param url: The URL to send the POST request to.
+        :param json_data: The JSON data to include in the POST request.
+        """
+        try:
+            response = self.post(
+                url,
```

Comment:
Concatenate `self.url.netloc` and `self.url_path` to get the full URL to post to here.